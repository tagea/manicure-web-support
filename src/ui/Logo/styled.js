import styled from 'styled-components';
import { Typography, Grid } from '@material-ui/core';

export const LogoStyled = styled(Grid)`
  &&{
    padding: 20px 0;
    padding-right:10px;
  }
`;

export const ImgStyled = styled('img')`
  vertical-align: middle;
`;

export const TopHeaderStyled = styled(Typography)`
  &&{
    font-family: 'Baloo Bhai', cursive;
    font-size: 2.3em;         
  }
`;

export const SubHeaderStyled = styled(Typography)`
  &&{
    font-weight: normal;
    font-size: 1.5em;
  }
`;
