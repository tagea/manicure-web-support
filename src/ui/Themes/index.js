import { createMuiTheme } from '@material-ui/core/styles';

export const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    fontSize: 16,
    fontFamily: [
      '"Roboto"',
      'sans-serif'
    ].join(',')
  },
  spacing: {
    unit: '0.2'
  },
  palette: {
    primary: {
      main: 'rgb(47, 128, 237)'
    },
    secondary: {
      main: 'rgb(255, 222, 107)',
      dark: 'rgb(234, 189, 32)'
    },
    surface: {
      main: 'rgb(255,255,255)',
      light: 'rgb(245, 245, 245)',
      transparent: 'rgba(0,0,0,0.08)'
    },
    text: {
      light: 'rgb(255,255,255)',
      main: 'rgb(0,0,0)',
      transparent: 'rgba(0,0,0,0.6)',
      error: 'rgb(217, 0, 0)'
    },
    shadow: {
      transparent: 'rgba(119, 119, 119, 0.15)',
      main: 'rgba(119, 119, 119, 0.5)'
    },
    border: {
      light: 'rgba(10,10,10,0.15)',
      main: 'rgb(47, 128, 237)',
      error: 'rgb(217, 0, 0)',
      success: 'rgb(39, 202, 25)',
      transparent: 'rgba(0,0,0,0)'
    }
  }
});
