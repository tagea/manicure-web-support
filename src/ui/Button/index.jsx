import React from 'react';
import PropTypes from 'prop-types';

import { ButtonStyled } from './styled';

const Button = (props) => {
  const { size, color, text, variant, handleClick } = props;
  return (
    <ButtonStyled
      size={size}
      color={color}
      variant={variant}
      onClick={handleClick}
    >
      {text}
    </ButtonStyled>
  );
};

Button.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  text: PropTypes.string,
  handleClick: PropTypes.func,
  variant: PropTypes.string
};

Button.defaultProps = {
  size: 'medium',
  color: 'secondary',
  text: '',
  handleClick: () => false,
  variant: 'text'
};

export { Button };
