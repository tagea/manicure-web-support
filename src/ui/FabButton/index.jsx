import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';

import { } from './styled';

const FabButton = (props) => {
  const { src, label, size, handleClick } = props;
  return (
    <Fab aria-label={label} size={size} color="secondary" onClick={handleClick}>
      <img src={src} alt={label} width="17" height="17" />
    </Fab>
  );
};

FabButton.propTypes = {
  src: PropTypes.string,
  label: PropTypes.string,
  size: PropTypes.string,
  handleClick: PropTypes.func
};

FabButton.defaultProps = {
  src: '',
  label: '',
  size: 'medium',
  handleClick: () => false
};

export { FabButton };
