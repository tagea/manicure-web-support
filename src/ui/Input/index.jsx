import React from 'react';
import PropTypes from 'prop-types';

import { InputStyled } from './styled';

const Input = props => (
  <InputStyled
    {...props}
    margin="dense"
    variant="outlined"
    fullWidth
  />
);

Input.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  error: PropTypes.bool,
  autoComplete: PropTypes.string,
  size: PropTypes.string
};

Input.defaultProps = {
  id: '',
  label: '',
  name: '',
  type: '',
  error: false,
  autoComplete: '',
  size: 'small'
};

export { Input };
