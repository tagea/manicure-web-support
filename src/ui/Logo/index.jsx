import React from 'react';
import Grid from '@material-ui/core/Grid';
import LogoSrc from '../../img/logo.png';

import { LogoStyled, ImgStyled, TopHeaderStyled, SubHeaderStyled } from './styled';

const Logo = () => (
  <LogoStyled container justify="center" alignItems="center" spacing={8}>
    <Grid item>
      <ImgStyled src={LogoSrc} alt="logo" width="58" height="58" />
    </Grid>
    <Grid item>
      <TopHeaderStyled variant="h3" align="center">ManiCure</TopHeaderStyled>
      <SubHeaderStyled variant="h4" align="center">Support</SubHeaderStyled>
    </Grid>
  </LogoStyled>
);

export { Logo };
