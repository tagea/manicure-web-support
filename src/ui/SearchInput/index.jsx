import React from 'react';
import { SearchStyled, SearchIconStyled, InputStyled } from './styled';

const SearchInput = () => (
  <SearchStyled>
    <InputStyled placeholder="Searchâ€¦" />
    <SearchIconStyled />
  </SearchStyled>
);

export { SearchInput };
