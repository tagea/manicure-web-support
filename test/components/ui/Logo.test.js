import React from 'react';
import { shallow } from 'enzyme';
import { Logo } from '../../../src/ui/Logo';
import {
  LogoStyled,
  ImgStyled,
  TopHeaderStyled,
  SubHeaderStyled
} from '../../../src/ui/Logo/styled';

it('render Logo styled-component with', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(LogoStyled)).toMatchSnapshot();
});

it('should render Logo with justify=center', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(LogoStyled).prop('justify')).toBe('center');
});

it('should render Logo with alignItems=center', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(LogoStyled).prop('alignItems')).toBe('center');
});

it('should render Logo with spacing=8', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(LogoStyled).prop('spacing')).toBe(8);
});

it('should render ImgStyled with not empty prop src', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(ImgStyled).prop('src')).not.toEqual('');
});

it('should render ImgStyled with not empty prop alt', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(ImgStyled).prop('alt')).not.toEqual('');
});

it('should render ImgStyled with not empty prop Width', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(ImgStyled).prop('width')).not.toEqual('');
});

it('should render ImgStyled with not empty prop Height', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(ImgStyled).prop('height')).not.toEqual('');
});

it('should render TopHeaderStyled with variant=h3', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(TopHeaderStyled).prop('variant')).toBe('h3');
});

it('should render TopHeaderStyled with align = |left|right|center', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(TopHeaderStyled).prop('align')).toMatch(/left|center|right/);
});

it('should render TopHeaderStyled with text', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(TopHeaderStyled).text().length).toBeGreaterThan(2);
});

it('should render SubHeaderStyled with variant=h3', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(SubHeaderStyled).prop('variant')).toBe('h4');
});

it('should render SubHeaderStyled with align = |left|right|center', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(SubHeaderStyled).prop('align')).toMatch(/left|center|right/);
});

it('should render SubHeaderStyled with text', () => {
  const LogoComponent = shallow(<Logo />);
  expect(LogoComponent.find(SubHeaderStyled).text().length).toBeGreaterThan(2);
});
