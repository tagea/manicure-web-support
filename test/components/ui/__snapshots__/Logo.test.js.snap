// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Logo styled-component with 1`] = `
<styled__LogoStyled
  alignItems="center"
  container={true}
  justify="center"
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
  >
    <styled__ImgStyled
      alt="logo"
      height="58"
      src="test-file-stub"
      width="58"
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <styled__TopHeaderStyled
      align="center"
      variant="h3"
    >
      ManiCure
    </styled__TopHeaderStyled>
    <styled__SubHeaderStyled
      align="center"
      variant="h4"
    >
      Support
    </styled__SubHeaderStyled>
  </WithStyles(Grid)>
</styled__LogoStyled>
`;
